### PR TITLE
EE movement followup for Standard Answers

### PR DESCRIPTION
**Pull request by trial-danswer**
## Description
This PR continues moving Standard Answers to the enterprise edition licensed portion of the repository.

Since APIs are already gated behind the EE flag, this is primarily a movement for future code licensing.

## How Has This Been Tested?
Locally to confirm that functionality still works.

## Accepted Risk
Same as prior EE migration - non-EE usage of these APIs will break.

## Related Issue(s)
#2378

## Checklist:
- [ ] All of the automated tests pass
- [ ] All PR comments are addressed and marked resolved
- [ ] If there are migrations, they have been rebased to latest main
- [ ] If there are new dependencies, they are added to the requirements
- [ ] If there are new environment variables, they are added to all of the deployment methods
- [ ] If there are new APIs that don't require auth, they are added to PUBLIC_ENDPOINT_SPECS
- [ ] Docker images build and basic functionalities work
- [ ] Author has done a final read through of the PR right before merge

### CODE REVIEW

File: backend/danswer/db/slack_bot_config.py [hunk 1]
```diff
@@ -14,8 +15,12 @@
 from danswer.db.persona import get_default_prompt
 from danswer.db.persona import mark_persona_as_deleted
 from danswer.db.persona import upsert_persona
-from danswer.db.standard_answer import fetch_standard_answer_categories_by_ids
 from danswer.search.enums import RecencyBiasSetting
+from danswer.utils.errors import EERequiredError
+from danswer.utils.variable_functionality import (
+    fetch_versioned_implementation_with_fallback,
+)
+from ee.danswer.db.standard_answer import fetch_standard_answer_categories_by_ids
```
Comment (yuhongsun96) [inline backend/danswer/db/slack_bot_config.py]: We never import anything from ee into MIT

Comment (trial-danswer) [reply backend/danswer/db/slack_bot_config.py]: Ah yep there's a case that still pulled this in, fixed.

File: backend/ee/danswer/danswerbot/slack/handlers/handle_standard_answers.py [hunk 2]
```diff
@@ -21,11 +21,11 @@
 from danswer.db.models import Prompt
 from danswer.db.models import SlackBotConfig
 from danswer.db.models import StandardAnswer as StandardAnswerModel
```
Comment (yuhongsun96) [inline backend/ee/danswer/danswerbot/slack/handlers/handle_standard_answers.py:23]: Often we'll call it DbStandardAnswer and the other one will be just StandardAnswer for the server model

Comment (trial-danswer) [reply backend/ee/danswer/danswerbot/slack/handlers/handle_standard_answers.py:23]: `StandardAnswerModel` is the original naming from this file before I touched it, do you want me to change it as part of this refactor?

**Review by yuhongsun96 — APPROVED** [review]
lgtm

File: backend/ee/danswer/danswerbot/slack/handlers/handle_standard_answers.py (hunk 2 above)
Comment (yuhongsun96) [reply backend/ee/danswer/danswerbot/slack/handlers/handle_standard_answers.py:23]: Doesn't matter, nobody will get confused by this anyway

File: backend/danswer/db/slack_bot_config.py (hunk 1 above)
Comment (yuhongsun96) [reply backend/danswer/db/slack_bot_config.py]: awesome thanks

**Pull request merged**